sympy/vector/integrals.py

```diff
@@ -0,0 +1,141 @@
+    >>> ParametricIntegral(C.z, semisphere)
+    8*pi
+
+    >>> ParametricIntegral(C.j - C.k, ParametricRegion((r, theta), (r*cos(theta), r*sin(theta))))
+    ParametricIntegral(C.j - C.k, ParametricRegion((r, theta), (r*cos(theta), r*sin(theta))))-
+
+    """
+    
+    def __new__(cls, field, parametricregion):
+        obj = super().__new__(cls, field, parametricregion)
+        return obj
+
+    def eval(self):
+        parametricregion = self.parametricregion
+        field = self.field
+        
+        coord_sys = _get_coord_sys_from_expr(field)
+        
+        if len(coord_sys) > 1:
+            raise ValueError
+        
+        coord_sys = next(iter(coord_sys))
+        base_vectors = coord_sys.base_vectors()
+        base_scalars = coord_sys.base_scalars()
+
+        parametricfield = field
+
+        r = Vector.zero
+        for i in range(len(parametricregion.definition)):
+            r += base_vectors[i]*parametricregion.definition[i]
+    
+        for i in range(len(parametricregion.definition)):
+            parametricfield = parametricfield.subs(base_scalars[i], parametricregion.definition[i])
+
+        if parametricregion.dimension == 1:
+            parameter = parametricregion.parameters[0]
+
+            r_diff = diff(r, parameter)        
+
+            lower, upper = parametricregion.limits[parameter][0], parametricregion.limits[parameter][1]
+            
+            if isinstance(parametricfield, Vector):
+                result = integrate(r_diff.dot(parametricfield), (parameter, lower, upper))
+            else:
+                result = integrate(r_diff.magnitude()*parametricfield, (parameter, lower, upper))
+            
+            if not isinstance(result, Integral):
+                return result
+                
+        elif parametricregion.dimension == 2:
+            u, v = parametricregion.parameters[0], parametricregion.parameters[1]
+            
+            r_u = diff(r, u)
+            r_v = diff(r, v)
+            normal_vector = simplify(r_u.cross(r_v))
+            
+            lower_u, upper_u = parametricregion.limits[u][0], parametricregion.limits[u][1]
+            lower_v, upper_v = parametricregion.limits[v][0], parametricregion.limits[v][1]
+            
+            if isinstance(parametricfield, Vector):
+                integrand = parametricfield.dot(normal_vector)
+            else:   
+                integrand = parametricfield*normal_vector.magnitude()
+            
+            integrand = simplify(integrand)
+            
+            case = self._bounds_case(u, v, parametricregion.limits)
+            
+            # Check the bounds to ascertain the order
+            # Case 1 : Both bounds are constants : can  perform integration in any order
+            # Case 2 : Bounds for u are in terms of v : integrate wrt to u first
+            # Case 3 : Bounds for v are in terms of u : integrate wrt to v first
+            
```

Comment:
Determining order for integration is necessary when bounds of parameters are interdependent.
Suppose we need to calculate area of a triangle:
```python
>>> integrate(1, (x, 0, 2), (y, 10 - 5*x))
20 - 10*x
>>> integrate(1, (y, 0, 10 - 5*x), (x, 0, 2))
10 # correct answer
```

Reply:
Topological sort?

Reply:
Yes, we need to perform a topological sort.
This function generates the graph and performs topological sort. It works on any number of parameters.
```python
    @classmethod             
    def _topological_sort_parameters(cls, parameters, limits):
    
        V = list(parameters)
        E = list()
        
        for p in parameters:
            lower_p = limits[p][0]
            upper_p = limits[p][1]
            
            lower_p = lower_p.atoms()
            upper_p = upper_p.atoms()
            for q in parameters:
                if p == q:
                    continue
                if lower_p.issuperset(set([q])) or upper_p.issuperset(set([q])):
                    E.append((p, q))   
        return topological_sort((V, E), key=default_sort_key)   
```
The only problem with this approach is when bounds are independent of each other, we can perform integration in any order. So if one order fails to calculate the integral, we can reverse the order and perform again. With my previous approach using cases, we can determine whether bounds are independent of each other but it works only for two parameters.

Reply:
What about calling the equation solver? The equation solver should be able to find a parametrization somehow.